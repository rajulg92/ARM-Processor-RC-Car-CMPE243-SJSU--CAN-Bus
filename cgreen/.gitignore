build
.idea
CMakeCache.txt
CMakeFiles
Testing
*.exe
*.stackdump
build-stamp
configure-stamp
*~
