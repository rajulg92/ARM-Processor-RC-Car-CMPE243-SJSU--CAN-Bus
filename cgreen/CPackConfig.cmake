# This file will be configured to contain variables for CPack. These variables
# should be set in the CMake list file of the project before CPack module is
# included. The list of available CPACK_xxx variables and their associated
# documentation may be obtained using
#  cpack --help-variable-list
#
# Some variables are common to all generators (e.g. CPACK_PACKAGE_NAME)
# and some are specific to a generator
# (e.g. CPACK_NSIS_EXTRA_INSTALL_COMMANDS). The generator specific variables
# usually begin with CPACK_<GENNAME>_xxxx.


SET(CPACK_BINARY_7Z "")
SET(CPACK_BINARY_BUNDLE "")
SET(CPACK_BINARY_CYGWIN "")
SET(CPACK_BINARY_DEB "")
SET(CPACK_BINARY_DRAGNDROP "")
SET(CPACK_BINARY_IFW "")
SET(CPACK_BINARY_NSIS "")
SET(CPACK_BINARY_OSXX11 "")
SET(CPACK_BINARY_PACKAGEMAKER "")
SET(CPACK_BINARY_RPM "")
SET(CPACK_BINARY_STGZ "")
SET(CPACK_BINARY_TBZ2 "")
SET(CPACK_BINARY_TGZ "")
SET(CPACK_BINARY_TXZ "")
SET(CPACK_BINARY_TZ "")
SET(CPACK_BINARY_WIX "")
SET(CPACK_BINARY_ZIP "")
SET(CPACK_CMAKE_GENERATOR "MinGW Makefiles")
SET(CPACK_COMPONENTS_ALL "")
SET(CPACK_COMPONENTS_ALL_IN_ONE_PACKAGE "1")
SET(CPACK_COMPONENT_APPLICATIONS_GROUP "Runtime")
SET(CPACK_COMPONENT_HEADERS_DEPENDS "libraries")
SET(CPACK_COMPONENT_HEADERS_DESCRIPTION "C/C++ header files for use with cgreen")
SET(CPACK_COMPONENT_HEADERS_DISPLAY_NAME "C/C++ Headers")
SET(CPACK_COMPONENT_HEADERS_GROUP "Development")
SET(CPACK_COMPONENT_LIBRARIES_DESCRIPTION "Libraries used to build programs which use cgreen")
SET(CPACK_COMPONENT_LIBRARIES_DISPLAY_NAME "Libraries")
SET(CPACK_COMPONENT_LIBRARIES_GROUP "Development")
SET(CPACK_COMPONENT_UNSPECIFIED_HIDDEN "TRUE")
SET(CPACK_COMPONENT_UNSPECIFIED_REQUIRED "TRUE")
SET(CPACK_GENERATOR "NSIS")
SET(CPACK_INSTALL_CMAKE_PROJECTS "C:/SJSU_Dev/cgreen;cgreen;ALL;/")
SET(CPACK_INSTALL_PREFIX "C:/Program Files (x86)/cgreen")
SET(CPACK_MODULE_PATH "C:/SJSU_Dev/cgreen/cmake/Modules")
SET(CPACK_NSIS_COMPRESSOR "/SOLID zlib")
SET(CPACK_NSIS_DISPLAY_NAME "The modern C/C++ unit testing/mocking framework")
SET(CPACK_NSIS_DISPLAY_NAME_SET "TRUE")
SET(CPACK_NSIS_INSTALLER_ICON_CODE "")
SET(CPACK_NSIS_INSTALLER_MUI_ICON_CODE "")
SET(CPACK_NSIS_INSTALL_ROOT "$PROGRAMFILES")
SET(CPACK_NSIS_MENU_LINKS "http://cgreen.sourceforge.net;Cgreen Homepage")
SET(CPACK_NSIS_PACKAGE_NAME "The modern C/C++ unit testing/mocking framework")
SET(CPACK_OUTPUT_CONFIG_FILE "C:/SJSU_Dev/cgreen/CPackConfig.cmake")
SET(CPACK_PACKAGE_DEFAULT_LOCATION "/")
SET(CPACK_PACKAGE_DESCRIPTION_FILE "C:/SJSU_Dev/cgreen/README.md")
SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "The modern C/C++ unit testing/mocking framework")
SET(CPACK_PACKAGE_FILE_NAME "cgreen-1.1.0beta-Windows32")
SET(CPACK_PACKAGE_INSTALL_DIRECTORY "cgreen")
SET(CPACK_PACKAGE_INSTALL_REGISTRY_KEY "cgreen")
SET(CPACK_PACKAGE_NAME "cgreen")
SET(CPACK_PACKAGE_RELOCATABLE "true")
SET(CPACK_PACKAGE_VENDOR "The CGreen Development Team")
SET(CPACK_PACKAGE_VERSION "1.1.0beta")
SET(CPACK_PACKAGE_VERSION_MAJOR "1")
SET(CPACK_PACKAGE_VERSION_MINOR "1")
SET(CPACK_PACKAGE_VERSION_PATCH "0")
SET(CPACK_PACKAGE_VERSION_STATUS "beta")
SET(CPACK_RESOURCE_FILE_LICENSE "C:/SJSU_Dev/cgreen/LICENSE")
SET(CPACK_RESOURCE_FILE_README "C:/Program Files (x86)/CMake/share/cmake-3.6/Templates/CPack.GenericDescription.txt")
SET(CPACK_RESOURCE_FILE_WELCOME "C:/Program Files (x86)/CMake/share/cmake-3.6/Templates/CPack.GenericWelcome.txt")
SET(CPACK_SET_DESTDIR "OFF")
SET(CPACK_SOURCE_7Z "")
SET(CPACK_SOURCE_CYGWIN "")
SET(CPACK_SOURCE_GENERATOR "TGZ")
SET(CPACK_SOURCE_IGNORE_FILES "~$;[.]swp$;/[.]svn/;/[.]git/;.gitignore;/build/;tags;cscope.*")
SET(CPACK_SOURCE_OUTPUT_CONFIG_FILE "C:/SJSU_Dev/cgreen/CPackSourceConfig.cmake")
SET(CPACK_SOURCE_PACKAGE_FILE_NAME "cgreen-1.1.0beta-src")
SET(CPACK_SOURCE_TBZ2 "")
SET(CPACK_SOURCE_TGZ "")
SET(CPACK_SOURCE_TXZ "")
SET(CPACK_SOURCE_TZ "")
SET(CPACK_SOURCE_ZIP "")
SET(CPACK_SYSTEM_NAME "win32")
SET(CPACK_TOPLEVEL_TAG "win32")
SET(CPACK_WIX_SIZEOF_VOID_P "4")

if(NOT CPACK_PROPERTIES_FILE)
  set(CPACK_PROPERTIES_FILE "C:/SJSU_Dev/cgreen/CPackProperties.cmake")
endif()

if(EXISTS ${CPACK_PROPERTIES_FILE})
  include(${CPACK_PROPERTIES_FILE})
endif()
