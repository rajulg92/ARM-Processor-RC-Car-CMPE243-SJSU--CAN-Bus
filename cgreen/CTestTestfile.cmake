# CMake generated Testfile for 
# Source directory: C:/SJSU_Dev/cgreen
# Build directory: C:/SJSU_Dev/cgreen
# 
# This file includes the relevant testing commands required for 
# testing this directory and lists subdirectories to be tested as well.
subdirs(doc)
subdirs(include)
subdirs(src)
