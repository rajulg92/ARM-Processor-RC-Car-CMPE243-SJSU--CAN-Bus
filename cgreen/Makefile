# CMAKE generated file: DO NOT EDIT!
# Generated by "MinGW Makefiles" Generator, CMake Version 3.6

# Default target executed when no arguments are given to make.
default_target: all

.PHONY : default_target

# Allow only one "make -f Makefile2" at a time, but pass parallelism.
.NOTPARALLEL:


#=============================================================================
# Special targets provided by cmake.

# Disable implicit rules so canonical targets will work.
.SUFFIXES:


# Remove some rules from gmake that .SUFFIXES does not remove.
SUFFIXES =

.SUFFIXES: .hpux_make_needs_suffix_list


# Suppress display of executed commands.
$(VERBOSE).SILENT:


# A target that is always out of date.
cmake_force:

.PHONY : cmake_force

#=============================================================================
# Set environment variables for the build.

SHELL = cmd.exe

# The CMake executable.
CMAKE_COMMAND = "C:\Program Files (x86)\CMake\bin\cmake.exe"

# The command to remove a file.
RM = "C:\Program Files (x86)\CMake\bin\cmake.exe" -E remove -f

# Escaping for special characters.
EQUALS = =

# The top-level source directory on which CMake was run.
CMAKE_SOURCE_DIR = C:\SJSU_Dev\cgreen

# The top-level build directory on which CMake was run.
CMAKE_BINARY_DIR = C:\SJSU_Dev\cgreen

#=============================================================================
# Targets provided globally by CMake.

# Special rule for the target package
package: preinstall
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Run CPack packaging tool..."
	"C:\Program Files (x86)\CMake\bin\cpack.exe" --config ./CPackConfig.cmake
.PHONY : package

# Special rule for the target package
package/fast: package

.PHONY : package/fast

# Special rule for the target package_source
package_source:
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Run CPack packaging tool for source..."
	"C:\Program Files (x86)\CMake\bin\cpack.exe" --config ./CPackSourceConfig.cmake C:/SJSU_Dev/cgreen/CPackSourceConfig.cmake
.PHONY : package_source

# Special rule for the target package_source
package_source/fast: package_source

.PHONY : package_source/fast

# Special rule for the target install/strip
install/strip: preinstall
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Installing the project stripped..."
	"C:\Program Files (x86)\CMake\bin\cmake.exe" -DCMAKE_INSTALL_DO_STRIP=1 -P cmake_install.cmake
.PHONY : install/strip

# Special rule for the target install/strip
install/strip/fast: install/strip

.PHONY : install/strip/fast

# Special rule for the target test
test:
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Running tests..."
	"C:\Program Files (x86)\CMake\bin\ctest.exe" --force-new-ctest-process $(ARGS)
.PHONY : test

# Special rule for the target test
test/fast: test

.PHONY : test/fast

# Special rule for the target rebuild_cache
rebuild_cache:
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Running CMake to regenerate build system..."
	"C:\Program Files (x86)\CMake\bin\cmake.exe" -H$(CMAKE_SOURCE_DIR) -B$(CMAKE_BINARY_DIR)
.PHONY : rebuild_cache

# Special rule for the target rebuild_cache
rebuild_cache/fast: rebuild_cache

.PHONY : rebuild_cache/fast

# Special rule for the target install
install: preinstall
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Install the project..."
	"C:\Program Files (x86)\CMake\bin\cmake.exe" -P cmake_install.cmake
.PHONY : install

# Special rule for the target install
install/fast: preinstall/fast
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Install the project..."
	"C:\Program Files (x86)\CMake\bin\cmake.exe" -P cmake_install.cmake
.PHONY : install/fast

# Special rule for the target install/local
install/local: preinstall
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Installing only the local directory..."
	"C:\Program Files (x86)\CMake\bin\cmake.exe" -DCMAKE_INSTALL_LOCAL_ONLY=1 -P cmake_install.cmake
.PHONY : install/local

# Special rule for the target install/local
install/local/fast: install/local

.PHONY : install/local/fast

# Special rule for the target edit_cache
edit_cache:
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Running CMake cache editor..."
	"C:\Program Files (x86)\CMake\bin\cmake-gui.exe" -H$(CMAKE_SOURCE_DIR) -B$(CMAKE_BINARY_DIR)
.PHONY : edit_cache

# Special rule for the target edit_cache
edit_cache/fast: edit_cache

.PHONY : edit_cache/fast

# Special rule for the target list_install_components
list_install_components:
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Available install components are: \"Unspecified\" \"headers\" \"libraries\""
.PHONY : list_install_components

# Special rule for the target list_install_components
list_install_components/fast: list_install_components

.PHONY : list_install_components/fast

# The main all target
all: cmake_check_build_system
	$(CMAKE_COMMAND) -E cmake_progress_start C:\SJSU_Dev\cgreen\CMakeFiles C:\SJSU_Dev\cgreen\CMakeFiles\progress.marks
	$(MAKE) -f CMakeFiles\Makefile2 all
	$(CMAKE_COMMAND) -E cmake_progress_start C:\SJSU_Dev\cgreen\CMakeFiles 0
.PHONY : all

# The main clean target
clean:
	$(MAKE) -f CMakeFiles\Makefile2 clean
.PHONY : clean

# The main clean target
clean/fast: clean

.PHONY : clean/fast

# Prepare targets for installation.
preinstall: all
	$(MAKE) -f CMakeFiles\Makefile2 preinstall
.PHONY : preinstall

# Prepare targets for installation.
preinstall/fast:
	$(MAKE) -f CMakeFiles\Makefile2 preinstall
.PHONY : preinstall/fast

# clear depends
depend:
	$(CMAKE_COMMAND) -H$(CMAKE_SOURCE_DIR) -B$(CMAKE_BINARY_DIR) --check-build-system CMakeFiles\Makefile.cmake 1
.PHONY : depend

#=============================================================================
# Target rules for targets named check

# Build rule for target.
check: cmake_check_build_system
	$(MAKE) -f CMakeFiles\Makefile2 check
.PHONY : check

# fast build rule for target.
check/fast:
	$(MAKE) -f CMakeFiles\check.dir\build.make CMakeFiles/check.dir/build
.PHONY : check/fast

#=============================================================================
# Target rules for targets named coverage

# Build rule for target.
coverage: cmake_check_build_system
	$(MAKE) -f CMakeFiles\Makefile2 coverage
.PHONY : coverage

# fast build rule for target.
coverage/fast:
	$(MAKE) -f CMakeFiles\coverage.dir\build.make CMakeFiles/coverage.dir/build
.PHONY : coverage/fast

#=============================================================================
# Target rules for targets named cgreen_shared

# Build rule for target.
cgreen_shared: cmake_check_build_system
	$(MAKE) -f CMakeFiles\Makefile2 cgreen_shared
.PHONY : cgreen_shared

# fast build rule for target.
cgreen_shared/fast:
	$(MAKE) -f src\CMakeFiles\cgreen_shared.dir\build.make src/CMakeFiles/cgreen_shared.dir/build
.PHONY : cgreen_shared/fast

# Help Target
help:
	@echo The following are some of the valid targets for this Makefile:
	@echo ... all (the default if no target is provided)
	@echo ... clean
	@echo ... depend
	@echo ... check
	@echo ... package
	@echo ... coverage
	@echo ... package_source
	@echo ... install/strip
	@echo ... test
	@echo ... rebuild_cache
	@echo ... install
	@echo ... install/local
	@echo ... edit_cache
	@echo ... list_install_components
	@echo ... cgreen_shared
.PHONY : help



#=============================================================================
# Special targets to cleanup operation of make.

# Special rule to run CMake to check the build system integrity.
# No rule that depends on this can have commands that come from listfiles
# because they might be regenerated.
cmake_check_build_system:
	$(CMAKE_COMMAND) -H$(CMAKE_SOURCE_DIR) -B$(CMAKE_BINARY_DIR) --check-build-system CMakeFiles\Makefile.cmake 0
.PHONY : cmake_check_build_system

