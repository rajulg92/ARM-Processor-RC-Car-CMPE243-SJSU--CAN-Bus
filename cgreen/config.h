/* Name of package */
#define PACKAGE "cgreen"

/* Version number of package */
#define VERSION "1.1.0beta"

/* #undef LOCALEDIR */
/* #undef DATADIR */
/* #undef LIBDIR */
#define PLUGINDIR "-1"
/* #undef SYSCONFDIR */
#define BINARYDIR "C:/SJSU_Dev/cgreen"
#define SOURCEDIR "C:/SJSU_Dev/cgreen"

/************************** HEADER FILES *************************/


/*************************** FUNCTIONS ***************************/


/*************************** LIBRARIES ***************************/


/**************************** OPTIONS ****************************/

/* Define if building with gcov instrumentation */
/* #undef CGREEN_INTERNAL_WITH_GCOV */

