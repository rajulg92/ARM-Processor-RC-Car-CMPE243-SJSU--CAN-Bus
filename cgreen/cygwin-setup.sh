#! /bin/sh
#
# I'm making some notes here for future reference: CYGWIN is *not*
# WIN32, but it *is* a DLL-platform So it loads DLL:s the same way
# Windows does, namely from the path. That means that for tests to run
# using the freshly build DLL:s in the build tree both include paths
# but also the PATH need to be prepended by the directory of the
# DLL. This is done with in the Cmake setup something like:
#
#   set_tests_properties(SomeTest PROPERTIES ENVIRONMENT "PATH=c:\somedir;c:\otherdir")
#
# which I found at http://stackoverflow.com/a/1079072/204658. The
# thing to work out is what CMAKE variables point to that
# directory.
#
# The follwing sets up the current shell in the same way, so that a
# CYGWIN developer could just source this script and then forget about
# it.
#

# Ensure that the script is sourced:
if [[ $_ == $0 ]]; then
   echo "You should really source this, like in '. cygwin-setup.sh'"
fi

# And here's the meat given that you have the standard build tree
export PATH="$PWD/build/src":"$PWD/build/tools":$PATH
