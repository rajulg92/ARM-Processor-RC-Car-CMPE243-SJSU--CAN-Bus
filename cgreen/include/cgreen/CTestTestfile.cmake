# CMake generated Testfile for 
# Source directory: C:/SJSU_Dev/cgreen/include/cgreen
# Build directory: C:/SJSU_Dev/cgreen/include/cgreen
# 
# This file includes the relevant testing commands required for 
# testing this directory and lists subdirectories to be tested as well.
subdirs(internal)
