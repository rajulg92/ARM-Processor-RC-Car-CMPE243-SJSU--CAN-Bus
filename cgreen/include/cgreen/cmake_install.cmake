# Install script for directory: C:/SJSU_Dev/cgreen/include/cgreen

# Set the install prefix
if(NOT DEFINED CMAKE_INSTALL_PREFIX)
  set(CMAKE_INSTALL_PREFIX "C:/Program Files (x86)/cgreen")
endif()
string(REGEX REPLACE "/$" "" CMAKE_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}")

# Set the install configuration name.
if(NOT DEFINED CMAKE_INSTALL_CONFIG_NAME)
  if(BUILD_TYPE)
    string(REGEX REPLACE "^[^A-Za-z0-9_]+" ""
           CMAKE_INSTALL_CONFIG_NAME "${BUILD_TYPE}")
  else()
    set(CMAKE_INSTALL_CONFIG_NAME "")
  endif()
  message(STATUS "Install configuration: \"${CMAKE_INSTALL_CONFIG_NAME}\"")
endif()

# Set the component getting installed.
if(NOT CMAKE_INSTALL_COMPONENT)
  if(COMPONENT)
    message(STATUS "Install component: \"${COMPONENT}\"")
    set(CMAKE_INSTALL_COMPONENT "${COMPONENT}")
  else()
    set(CMAKE_INSTALL_COMPONENT)
  endif()
endif()

if("${CMAKE_INSTALL_COMPONENT}" STREQUAL "headers" OR NOT CMAKE_INSTALL_COMPONENT)
  file(INSTALL DESTINATION "${CMAKE_INSTALL_PREFIX}/include/cgreen" TYPE FILE FILES
    "C:/SJSU_Dev/cgreen/include/cgreen/assertions.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/boxed_double.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/breadcrumb.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/cdash_reporter.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/cgreen.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/constraint.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/constraint_syntax_helpers.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/cpp_assertions.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/cpp_constraint.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/cute_reporter.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/legacy.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/mocks.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/string_comparison.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/reporter.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/runner.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/suite.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/text_reporter.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/unit.h"
    "C:/SJSU_Dev/cgreen/include/cgreen/vector.h"
    )
endif()

if(NOT CMAKE_INSTALL_LOCAL_ONLY)
  # Include the install script for each subdirectory.
  include("C:/SJSU_Dev/cgreen/include/cgreen/internal/cmake_install.cmake")

endif()

