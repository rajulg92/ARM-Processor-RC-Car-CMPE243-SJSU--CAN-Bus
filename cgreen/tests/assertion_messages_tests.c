#include <cgreen/cgreen.h>
#include <cgreen/constraint.h>

#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

#ifdef __cplusplus
using namespace cgreen;
#endif

Describe(AssertionMessage);
BeforeEach(AssertionMessage) {}
AfterEach(AssertionMessage) {}


char something[42];
Ensure(AssertionMessage, for_comparing_content_with_negative_length) {
    assert_that(47, is_equal_to_contents_of(something, -4));
}

Ensure(AssertionMessage, return_value_constraints_are_not_allowed) {
    assert_that(0, will_return(1));
}

Ensure(AssertionMessage, for_compare_area_to_null) {
    char area[100];
    void *null_pointer = NULL;
    assert_that(area, is_equal_to_contents_of(null_pointer, 1));
}

Ensure(AssertionMessage, for_compare_null_to_area) {
    char area[100];
    assert_that(NULL, is_equal_to_contents_of(area, 1));
}
