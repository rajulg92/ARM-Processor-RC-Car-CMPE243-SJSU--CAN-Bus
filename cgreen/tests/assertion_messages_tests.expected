Running "assertion_messages_tests" (4 tests)...
assertion_messages_tests.c: Failure: AssertionMessage -> for_compare_area_to_null 
	Wanted to compare contents of [null_pointer] but it had a value of NULL.
		If you want to explicitly check for null, use the is_null constraint instead.

assertion_messages_tests.c: Failure: AssertionMessage -> for_compare_null_to_area 
	Wanted to compare contents with [area],
		but NULL was used for the pointer we wanted to compare to.
		If you want to explicitly check for null, use the is_null constraint instead.

assertion_messages_tests.c: Failure: AssertionMessage -> for_comparing_content_with_negative_length 
	Wanted to compare contents with [something],
		but [-4] was given for the comparison size.

assertion_messages_tests.c: Failure: AssertionMessage -> return_value_constraints_are_not_allowed 
	Got constraint of type [return value],
		but they are not allowed for assertions, only in mock expectations.

Completed "AssertionMessage": 4 failures in 0ms.
Completed "assertion_messages_tests": 4 failures in 0ms.
